File: solutions/go/resistor-color-duo/1/resistor_color_duo.go
package resistorcolorduo

import "strings"

func Value(colors []string) int {
	colorMap := map[string]int{
		"black":  0,
		"brown":  1,
		"red":    2,
		"orange": 3,
		"yellow": 4,
		"green":  5,
		"blue":   6,
		"violet": 7,
		"grey":   8,
		"white":  9,
	}

	if len(colors) < 2 {
		return 0
	}

	firstDigit := colorMap[strings.ToLower(colors[0])]
	secondDigit := colorMap[strings.ToLower(colors[1])]

	return firstDigit*10 + secondDigit
}

File: solutions/julia/resistor-color-duo/1/resistor-color-duo.jl
const COLOR_MAP = Dict(
    "black"  => 0,
    "brown"  => 1,
    "red"    => 2,
    "orange" => 3,
    "yellow" => 4,
    "green"  => 5,
    "blue"   => 6,
    "violet" => 7,
    "grey"   => 8,
    "white"  => 9,
)

function colorcode(colors::Vector{String})
    
    first_digit = COLOR_MAP[colors[1]]
    
    second_digit = COLOR_MAP[colors[2]]

    
    return first_digit * 10 + second_digit
end

function COLORS()
    
    
    
    pairs = collect(COLOR_MAP)
    
    
    sort!(pairs, by = x -> x[2])
    
    
    return [p[1] for p in pairs]
end
